Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+Find.swift

```diff
@@ -480,17 +487,13 @@ private extension PathHierarchy.Node {
             return true
         }
         // Otherwise, check if the node's symbol matches the provided disambiguation
-        else if let symbol = symbol {
-            guard name == component.name else {
-                return false
+        else if let symbol = symbol, let disambiguation = component.disambiguation {
+            switch disambiguation {
```

Comment:
And here for example you might be able to avoid the switch statement and avoid referring to the enum case directly.